connector/src/main/java/org/geysermc/connector/network/translators/inventory/DoubleChestInventoryTranslator.java

```diff
@@ -40,60 +40,98 @@
+        String otherPairBlockId = blockId;
+        if (javaBlockId != null && !inventory.isCreatingNewBlock()) {
+            String isolatedBlockId = javaBlockId.split("\\[")[0];
+            if (isolatedBlockId.equals("minecraft:chest") || isolatedBlockId.equals("minecraft:trapped_chest")) {
+                newBlockId = BlockTranslator.getBedrockBlockId(BlockTranslator.getJavaBlockState(javaBlockId));
+                if (javaBlockId.contains("type=left")) {
+                    // Using the Java block ID, figure out where the other chest is located
+                    if (javaBlockId.contains("facing=north")) {
+                        pairPosition = position.add(Vector3i.UNIT_X);
+                    } else if (javaBlockId.contains("facing=south")) {
+                        pairPosition = position.sub(Vector3i.UNIT_X);
+                    } else if (javaBlockId.contains("facing=east")) {
+                        pairPosition = position.add(Vector3i.UNIT_Z);
+                    } else if (javaBlockId.contains("facing=west")) {
+                        pairPosition = position.sub(Vector3i.UNIT_Z);
+                    }
+                    otherPairBlockId = javaBlockId.replace("type=left", "type=right");
+                } else if (javaBlockId.contains("type=right")) {
+                    if (javaBlockId.contains("facing=north")) {
+                        pairPosition = position.sub(Vector3i.UNIT_X);
+                    } else if (javaBlockId.contains("facing=south")) {
+                        pairPosition = position.add(Vector3i.UNIT_X);
+                    } else if (javaBlockId.contains("facing=east")) {
+                        pairPosition = position.sub(Vector3i.UNIT_Z);
+                    } else if (javaBlockId.contains("facing=west")) {
+                        pairPosition = position.add(Vector3i.UNIT_Z);
+                    }
+                    otherPairBlockId = javaBlockId.replace("type=right", "type=left");
+                } else {
+                    // Chest type is single
+                    inventory.setCreatingNewBlock(true);
+                }
+            } else {
+                inventory.setCreatingNewBlock(true);
+            }
+        } else {
+            inventory.setCreatingNewBlock(true);
+        }
+
+        if (inventory.isCreatingNewBlock()) {
+            // Reset position to player because this is the wrong block
+            position = session.getPlayerEntity().getPosition().toInt().add(Vector3i.UP);
+            pairPosition = position.add(Vector3i.UNIT_X);
+        }
 
         UpdateBlockPacket blockPacket = new UpdateBlockPacket();
         blockPacket.setDataLayer(0);
         blockPacket.setBlockPosition(position);
-        blockPacket.setRuntimeId(blockId);
+        blockPacket.setRuntimeId(inventory.isCreatingNewBlock() ? BlockTranslator.getBedrockBlockId(BlockTranslator.getJavaBlockState(blockId)) : newBlockId);
```

Comment:
Your getting the blockstate from the bedrock id to turn it back into the bedrock id, couldn't this be simplified to become:
```java
blockPacket.setRuntimeId(inventory.isCreatingNewBlock() ? blockId) : newBlockId);
```